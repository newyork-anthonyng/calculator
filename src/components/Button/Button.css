.Button {
	color: white;
	font-size: 24px;
	display: flex;
	align-items: center;
	justify-content: center;
	height: 80px;
	width: 25%;
	margin: 2px;
}
