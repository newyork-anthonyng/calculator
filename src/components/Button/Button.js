import React, { PropTypes } from 'react';
import './Button.css';

const Button = ({ text, onClick, style }) => (
	<div
		style={style}
		className="Button"
		onClick={() => onClick(text)}
	>
		{text}
	</div>
);

Button.propTypes = {
	text: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	style: PropTypes.object
};

export default Button;
