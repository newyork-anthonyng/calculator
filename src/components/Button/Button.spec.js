import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
	it('should render', () => {
		const component = renderer.create(
			<Button
				text="+"
				onClick={() => {}}
				style={{}}
			/>
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should run callback when clicked', () => {
		const cb = jest.fn();
		const wrapper = shallow(
			<Button
				text="foo"
				onClick={cb}
				style={{}}
			/>
		);

		wrapper.simulate('click');
		expect(cb.mock.calls[0][0]).toEqual('foo');
	});

	it('should accept a style object', () => {
		const style = {
			backgroundColor: 'yellow'
		};
		const wrapper = shallow(
			<Button
				text="foo"
				onClick={() => {}}
				style={style}
			/>
		);

		const buttonStyle = wrapper.prop('style');
		expect(buttonStyle).toEqual(style);
	});
});
