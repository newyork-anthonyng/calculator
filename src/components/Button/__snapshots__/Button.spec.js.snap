exports[`Button should render 1`] = `
<div
  className="Button"
  onClick={[Function]}
  style={Object {}}>
  +
</div>
`;
